packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -63,6 +69,32 @@ const extractAttachedFilesMetadata = (metadata: Record<string, string>): FileMet
   }
 };
 
+/* @conditional-compile-remove(dlp) */
+const convertToUiBlockedMessage = (
+  message: ChatMessageWithStatus,
+  userId: string,
+  isSeen: boolean,
+  isLargeGroup: boolean
+): BlockedMessage => {
+  const messageSenderId = message.sender !== undefined ? toFlatCommunicationIdentifier(message.sender) : userId;
+  return {
+    messageType: 'blocked',
+    createdOn: message.createdOn,
+    content: message.content?.message,
+    contentType: sanitizedMessageContentType(message.type),
+    status: !isLargeGroup && message.status === 'delivered' && isSeen ? 'seen' : message.status,
+    senderDisplayName: message.senderDisplayName,
+    senderId: messageSenderId,
+    messageId: message.id,
+    clientMessageId: message.clientMessageId,
+    editedOn: message.editedOn,
+    deletedOn: message.deletedOn,
+    mine: messageSenderId === userId,
+    metadata: message.metadata,
```

Comment:
DLP message is just like a regular message, it can also be deleted (so it will completely not be shown and not just the content is removed). But will remove the metadata and contentType